frontend/app/src/scenes/Home/scenes/AddSubCategory/index.tsx

```diff
@@ -90,47 +134,62 @@ class AddSubCategory extends React.Component<
+                    placeholder="Please select a category"
+                  >
+                    {this.state.categories.map((category: Category) => {
+                      return (
+                        <Option key={category.id} value={category.id}>
+                          {category.name}
+                        </Option>
+                      );
+                    })}
+                  </Select>
+                </Form.Item>
+                <Title level={3}>Name</Title>
+                <Text>Name of the subcategory</Text>
+                <Form.Item name="name" className={mainStyles.formItem}>
+                  <Input placeholder="ex: Local Curriculum" />
+                </Form.Item>
+                <Form.Item>
+                  <Button type="primary" htmlType="submit">
+                    Save
+                  </Button>
+                </Form.Item>
+              </Form>
+            </Col>
+          </Row>
+        </Spin>
+      </Col>
+    );
+  };
   render() {
     return (
       <div>
         <Row className={mainStyles.innerContent}>
-          <Col lg={24} xl={{ span: 20, offset: 2 }}>
-            <Title>Add a new Subcategory</Title>
-            <Row>
-              <Col md={12}>
-                <Spin tip="Loading..." spinning={this.state.isLoading}>
-                  <Form size="large" onFinish={this.onFinish}>
-                    <Title level={3}>Category</Title>
-                    <Text>Select a category</Text>
-                    <Form.Item name="category" className={mainStyles.formItem}>
-                      <Select
-                        className={mainStyles.formSelect}
-                        placeholder="Please select a category"
-                      >
-                        {this.state.categories.map((category: Category) => {
-                          return (
-                            <Option key={category.id} value={category.id}>
-                              {category.name}
-                            </Option>
-                          );
-                        })}
-                      </Select>
-                    </Form.Item>
-                    <Title level={3}>Name</Title>
-                    <Text>Name of the subcategory</Text>
-                    <Form.Item name="name" className={mainStyles.formItem}>
-                      <Input placeholder="ex: Local Curriculum" />
-                    </Form.Item>
-                    <Form.Item>
-                      <Button type="primary" htmlType="submit">
-                        Add Subcategory
-                      </Button>
-                    </Form.Item>
-                  </Form>
-                </Spin>
-              </Col>
-            </Row>
-          </Col>
+          {this.state.subcategory != null &&
+            this.subCategoryForm('Edit Subcategory')}
+          {this.state.subcategory == null &&
+            this.subCategoryForm('Add a new Subcategory')}
```

Comment:
What about creating a new string to decide the title?

Instead of passing as a prop for a functional component?

Reply:
Also, check the component type instead of null check

Reply:
the component type is set before getting the payload so it won’t work for the type

Reply:
fixed